ID3v2 key notation export

daschuer opened the pull request:
This makes sure that newly edited keys are written with the ID3v2 format to the database and track metadata. 
Before the current selected key notation from the preferences is used. 
This is likely incompatible to other tools. 

Fixes #10890

This PR is on top of https://github.com/mixxxdj/mixxx/pull/11096

flosse committed:
Denon MC7000: Use parameter buttons for beat jumps

fwcd committed:
Add optional jog wheel acceleration to MC7000 mapping

The acceleration is disabled by default.

daschuer committed:
Read and write keyText to database, independent from the key format in preferences for GUI

daschuer committed:
Keep keyText imported from metadata unchanged. It may has extra info like tuning ore advanced modes we cannot parse.

daschuer committed:
Remove global key functions form TrackRecord and inline them in Track. This is more efficient and easier to understand.

daschuer committed:
Improve key code

daschuer committed:
Adding a test that verifies that the metadata key field is kept if the user does not edit it.

daschuer committed:
Add preferred notation form ID3v2

daschuer commented:
My original idea was to make this a part of 2.3.4. 
But it has grown a bit because of the required refactoring as a prerequisite.  
Now I am unsure. This is an important fix, to write the correct format and it has a unit-test.
Ideas?

uklotzde commented on src/library/dao/trackdao.cpp:
```diff
@@ -1313,9 +1308,7 @@ TrackPointer TrackDAO::getTrackById(TrackId trackId) const {
-            if ((*populator)(queryRecord, i, pTrack)) {
-                shouldDirty = true;
-            }
+            shouldDirty = (*populator)(queryRecord, i, pTrack);
```
Why did you change this unrelated code and introduce a bug? It no longer works as intended.

daschuer replied:
Ah, OK, I see. I will revert this. Thank you.

uklotzde replied:
Please note that I discovered this flaw by chance and didn't do a review of the remaining code.

atskler commented:
After this fix what file formats and what kind of tags will be affected at the end?
For example an mp3 with ape tag will get the new notation format? And mp4 tags? flac tags?
For general consistency and uniformity should I rewrite tags for all my files?
Or just specific formats with specific tags?

Will be mixed notation info in Mixxx's database or that will be updated to be uniform and consistent?

daschuer commented:
After merging this PR, Mixxx reads the key field from all tag types literally to the database. The GUI shows the key as you specify it in the key preferences. When you edit a key value or reanalyse the track, the key is stored with the ID3v2 notation for all tag types.   
This is a most conservative approach that keeps the file metadata unchanged along with possible extra infos like tuning or special modes. It does not fix the non-standard key notation written by the current Mixxx.  

Does that work for you?

atskler commented:
If i'm right all this started with "can cause issues in third party applications" (10890). And if we are talking about issues with third party applications, then I would like to have the option to "fix the non-standard key notation written by the current Mixxx" for them, additionally because I would not like to see mixed notation sourced from tags in a 3rd party application like foobar2000.

daschuer commented:
Yes, right. What is your preferred solution?

atskler commented:
Provide an option in preferences to choose:
- legacy notation scheme (idiosyncratic, less compatible)
- standardized notation scheme (fully compatible)

Also provide a function to rewrite all tags accordingly.

On new installations use the standardized scheme by default.
On existing installations use the legacy scheme by default.

What do you think?

daschuer commented:
I consider writing the legacy format a bug. So I don't think it is reasonable to provide an option for this format. Do you have a use case for this? 

We have currently no good way to mass update all Tracks. This will take loooong and we need to implement a reasonable feedback for this. Would it help to update only touched tracks? This way you could be forced by analysis or manual via the "Export Metadata to file tags".

There is by the way the risk to loose the tuning information written by Rapid Evolution.

atskler commented:
Then consider that on my side Mixxx implanted bug into 6072 files, and all those files needs to be fixed if we really have the fear regarding those lurking "issues in third party applications".

"Do you have a use case for this?" - Yes, a general player can't display this or that notation format, just show what is written into the tag, and for example for me the 12B, 9A format is more convenient. If you fix this, you actually brake my "way of life" and there is no way back.

Regarding the Rapid Evolution risk you mentioned: it sounds like it also uses some non-standard idiosyncratic "bug" format. Otherwise why could be any problem here? And if there is implemented compatibility for a Rapid Evolution "bug", then a backwards compatibility for the other "buggy" notation format may be also implemented. (Also I found that Rapid Evolution is discontinued, abandoned, unsupported software.)

At the and, in my opinion, you have to provide fix for the files also, not just for the code, if you really want to mend this bug, and that fear.

How do you see all this?

atskler commented:
Anyways as I see it currently Mixxx not reads/imports back the key info from the mp3 tags, at lest in the format of “12A”. Probably the proposed standard way should be implemented, and then the tags rewritten. But then the standard should be precisely followed regardless of the compatibility with same abandoned software that again does non-standard things, or a special compatibility mode should be provided for them in preferences, disabled by default.

daschuer commented:
Sounds good. I think that is nothing we should do in a stable branch. I will re-target it to 2.4.

fwcd committed:
clang-format: Indent Objective-C blocks with 4 spaces

ronso0 committed:
Library: center date values

fwcd commented:
Could you clarify which key notation you are writing here? Haven't dug into the code, but IIRC ID3v2 doesn't define a standard key notation (or even tag), so I'll assume that traditional notation (e.g. "D♭") is meant?

Commit:
Add track move commands to controls library. Sending key event to library widget.

Co-authored-by: ronso0 <ronso0@mixxx.org>

daschuer committed:
Revert "DEBUG output for fullscreen changes"

This reverts commit 5a4fe74ca9bd90814ef0eadf3730a073fd01783f.

ronso0 committed:
allow changing the waveform overview type without reloading the skin

ronso0 committed:
BpmControl: add `beats_undo_possible` helper

ronso0 committed:
LateNight: add `beats_undo_adjustment` button

daschuer committed:
Add CHANGLOG entries about raised OS requirements

ronso0 committed:
AutoDJ: add 'Clear Queue' menu action

ronso0 committed:
AutoDJ: use remove hotkey Del for Clear Queue and Remove Crate actions

ronso0 committed:
WOverview: move redundant drawing code to drawNextPixmapPart()

daschuer committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

ronso0 committed:
WOverview: use QStringLiteral for ScopedTimer id

ronso0 committed:
use enum class for waveform overview type

daschuer committed:
Update Translation template. Found 3104 source text(s) (0 new and 3104 already existing)

daschuer committed:
Pull latest translations from https://www.transifex.com/mixxx-dj-software/mixxxdj/mixxx2-5/. Compile QM files out of TS files that are used by the localized app

daschuer committed:
Merge pull request #13273 from ronso0/overview-type-integration

allow changing the waveform overview type without reloading the skin

daschuer committed:
Install missing build dependencies for Ubuntu Oracular

The packages libqt6labsqmlmodels6, libqt6quickcontrols2impl6 and libqt6quicklayouts6 have been integrated in the corrosponding qml6-module packages.

fwcd committed:
CMakeLists: Work around Qt shader bug with Xcode

fwcd committed:
AudioUnitManager: Disable unavailable in-process instantiation on iOS

ronso0 committed:
WOverview: use PollingControlProxy, remove ControlFlag::NoAssertIfMissing

daschuer committed:
Merge pull request #13379 from fwcd/fix-qt-shader-bug-xcode

CMakeLists: Work around Qt shader bug with Xcode

daschuer committed:
Merge pull request #13383 from fwcd/disable-audiounit-in-process-ios

AudioUnitManager: Disable unavailable in-process instantiation on iOS

ronso0 committed:
Tango: use qproperty-layoutSpacing for beatgrid buttons

ronso0 committed:
Tango: add `beats_undo_adjustment` button

ronso0 committed:
Deere: add `beats_undo_adjustment` button

ronso0 committed:
Merge pull request #13098 from nlw0/nic/track-move-controls

Add track move commands to controls library

acolombier committed:
Kontrol S4 Mk3: rework jogwheel speed compute and motorized platter

Swiftb0y committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

alhadebe committed:
add controller settings for the Numark Scratch

Swiftb0y committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

ronso0 committed:
(fix) Library: show Properties action in Missing and Hidden

JoergAtGithub committed:
Merge remote-tracking branch 'upstream/2.5' into SlowEventLoopNotify

JoergAtGithub committed:
Only start the event loop performance timer if we are in developer mode

JoergAtGithub committed:
Adjusted the warning threshold to 5ms and added a comment, which explains this

JoergAtGithub committed:
Merge pull request #13370 from ronso0/overview-type-enum

use enum class for waveform overview type

JoergAtGithub committed:
Merge pull request #13326 from ronso0/effects-visible-list

(fix) Effects: show newly added effects, read/write HiddenEffects

ronso0 committed:
AutoDJ: force-show decks 3/4 if we are going to use them

daschuer committed:
Merge pull request #13426 from ronso0/track-prop-in-hidden-and-missing

(fix) Library: show Properties action in Missing and Hidden

Commit:
replace relative paths to LateNight skin with absolute paths

ronso0 committed:
Library: fix font reset in multiline comment editor

daschuer committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

daschuer committed:
Update Translation template. Found XXXX source text(s) (0 new and 3104 already existing)

daschuer committed:
Pull latest translations from https://www.transifex.com/mixxx-dj-software/mixxxdj/mixxx2-5/. Compile QM files out of TS files that are used by the localized app

Swiftb0y committed:
Improve KeyUtils class

Swiftb0y committed:
Merge pull request #13404 from NotYourAverageAl/NumarkScratchSettings

feat(controllers): add controller settings for the Numark Scratch

Commit:
added combobox to limit until mark text height to 1/3rd of waveform viewer or full height

m0dB committed:
Merge pull request #13448 from ronso0/lib-comment-editor-font-fix

Library: fix font reset in multiline comment editor

daschuer committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

daschuer committed:
Pull latest translations from https://www.transifex.com/mixxx-dj-software/mixxxdj/mixxx2-5/. Compile QM files out of TS files that are used by the localized app

fwcd committed:
SlipModeShader: Set precision qualifiers for GL ES support

fwcd committed:
CmdLineArgs: Check TERM to determine whether to use colors

fwcd committed:
LegacySkinParser: Short-circuit if template fails to open

JoergAtGithub committed:
Merge pull request #13485 from fwcd/slip-mode-shader-precision-qualifiers

SlipModeShader: Set precision qualifiers for GL ES support

daschuer committed:
Merge pull request #13488 from fwcd/fix-legacy-skin-template-file-open-fallback

LegacySkinParser: Short-circuit if template fails to open

ronso0 committed:
LateNight: add/tweak CueDelete icons

Commit:
Make references to waveform.xml absolute to the LateNight theme and not relative

ronso0 committed:
Merge pull request #13501 from morgannunan/2.5-late-wave-fix

Update waveforms_container.xml

Holzhaus committed:
Merge branch '2.4' of github.com:mixxxdj/mixxx into 2.5

Holzhaus committed:
Merge branch '2.4' of github.com:mixxxdj/mixxx into 2.5

JoergAtGithub committed:
Merge pull request #13495 from ronso0/late-cue-delete

(fix) LateNight: add/tweak CueDelete icons

ronso0 committed:
Merge pull request #13507 from Holzhaus/merge-2.4-into-2.5

Merge 2.4 into 2.5

daschuer committed:
Update CHANGLOG with changes for 2.5

daschuer committed:
Remove duplicate enty of #13092

JoergAtGithub committed:
Merge pull request #13362 from daschuer/changelog_25

Add CHANGLOG entries about raised OS requirements

daschuer committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

daschuer committed:
Improve comments of key functions in Keys

daschuer committed:
Use preferred notation form ID3v2 when writing key changes to files and database

ronso0 committed:
(fix/Qt6) Skins: hide reserved space for invisible table header sort indicators

ronso0 committed:
Tango: don't change header sort arrow icon on hover

ronso0 committed:
(fix) use 'played' track color also for played/play count delegate

Add CheckboxDelegate base class, use it for BPMDelegate and play count delegate.
This allows moving redundant paintItem() code to CheckboxDelegate, and thereby
removing the PlayCountDelegate.

m0dB committed:
Merge pull request #13364 from ronso0/lib-sidebar-del-hotkey-autodj

(fix) Library: repair Del hotkey in sidebar, add Clear Queue action to AutoDJ, add hotkeys

JoergAtGithub committed:
Merge pull request #13538 from ronso0/played-delegate-color-fix

(fix) use 'played' track color also for played/play count delegate

JoergAtGithub committed:
Merge pull request #13535 from ronso0/skins-table-header-sort-bug

(fix/Qt6) Skins: hide reserved space for invisible table header sort indicators

ronso0 committed:
improve Taglib/SoundSource logging: FileType enum as string (char*), omit 'file://' prefix for local files

daschuer committed:
Fix returning the formated value

daschuer committed:
Keep detected key map in case the imported metadata value corresponds to it

daschuer committed:
Allow to delete the keys by removing the global key value.

daschuer committed:
Comment updateGlobalKeyNormalizeText()

ronso0 committed:
(fix) pre-commit/qsscheck: add c++ ObjectNames set via variable in #13538

Swiftb0y committed:
Merge pull request #13549 from ronso0/qss-fix

(fix) pre-commit/qsscheck: add c++ ObjectNames set via variable in #13538

daschuer committed:
Improve TrackMetadataExportTest

daschuer committed:
make newKeys const Track::replaceMetadataFromSource

daschuer committed:
Make use of FRIEND_TEST macro

To not rely on Google Test internas

Co-authored-by: Swiftb0y <12380386+Swiftb0y@users.noreply.github.com>

Swiftb0y committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

Swiftb0y committed:
Merge pull request #11096 from daschuer/gh11095

Keep the metadata key text unchanged, use it as the origin of information

Swiftb0y committed:
Merge branch '2.4' into 2.5

Swiftb0y committed:
chore: fix pre-commit

Swiftb0y committed:
Merge remote-tracking branch 'upstream/2.5' into 2.5

fwcd committed:
Add explanatory comment

Co-authored-by: JoergAtGithub <64457745+JoergAtGithub@users.noreply.github.com>

fwcd committed:
MC7000: Add user-configurable jogwheel settings

fwcd committed:
MC7000: Wire up user-configurable settings

fwcd committed:
MC7000: Update ranges for acceleration parameters

fwcd committed:
MC7000: Fix small stylistic typo

fwcd committed:
MC7000: Update sensitivity range

JoergAtGithub committed:
Merge pull request #13541 from ronso0/logging-taglib-soundsource

improve Taglib/SoundSource logging

ekordas committed:
MIDI for light: Implement new Active deck heuristic
Detects the active Deck by sorting the Decks by their volumes.
fixes #11322, closes #13138

fwcd committed:
MC7000: Fix typo

fwcd committed:
CmdLineArgs: Add alacritty to the list of terminals

JoergAtGithub committed:
Merge pull request #4684 from fwcd/mc7000-jog-acceleration

Denon MC7000: Add optional jog wheel acceleration to the controller mapping

daschuer committed:
Merge pull request #13486 from fwcd/disable-log-colors-ios

CmdLineArgs: Determine whether to color output based on `TERM` variable

fwcd committed:
Merge pull request #13513 from ekordas/MIDI-for-light-Implement-new-Active-deck-heuristic

MIDI for light: Implement new Active deck heuristic

ronso0 committed:
Merge remote-tracking branch 'mixxx/2.4' into 2.5

fwcd committed:
Denon MC7000: Remove unused paramButton variable

fwcd committed:
Denon MC7000: Use early return for better readability

fwcd committed:
Denon MC7000: Unify parameter button logic and make it customizable

fwcd committed:
Denon MC7000: Add intro/outro mode for parameter buttons

fwcd committed:
Denon MC7000: Add description for parameter button mode setting

fwcd committed:
Denon MC7000: Set starsAndColor as default param button mode

fwcd committed:
Denon MC7000: Make pitch play override of param buttons customizable

fwcd committed:
Denon MC7000: Use nullish-coalescing operator for param settings

ronso0 committed:
(fix) WStarRating: remove unneeded signal that may cause rating reset via track menu

ronso0 committed:
(fix/skins/Qt6) push skin setiings to the top

daschuer committed:
Merge pull request #13609 from ronso0/star-rating-reset-bug

(fix) WStarRating: remove unneeded signal that may cause rating reset via track menu

daschuer committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

ronso0 committed:
Merge remote-tracking branch 'mixxx/2.4' into 'mixxx/2.5'

Swiftb0y committed:
Merge pull request #13589 from fwcd/mc7000-parameter-buttons

Denon MC7000: Unify parameter button logic and add customizable modes

ronso0 committed:
WOverview: update immediately when the normalize option or Global gain changed

JoergAtGithub committed:
Introduce m_isDeveloper member variable in MixxxApplication to cache
developer mode status, reducing redundant calls to
CmdlineArgs::Instance().getDeveloper().

JoergAtGithub committed:
Merge pull request #13634 from ronso0/overview-normalize-proxy

WOverview: update immediately when the normalize option or Global gain changed

daschuer committed:
Fix writing global key text after analyzing to library and file metadata.

This fixes #13646 a regression since #11096

Swiftb0y committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

Swiftb0y committed:
fix: apply GroupFeatureState changes to glitcheffect to fix build

acolombier committed:
fix: prevent crash when no app windows have the focus

acolombier committed:
fix: allow controller setting integer to have a negative range

acolombier committed:
Add missing ptr marker and prefix

Swiftb0y committed:
Merge pull request #13657 from acolombier/fix/crash-windows-focus

fix: prevent crash when no app windows have the focus

acolombier committed:
Refactor setting check logic

acolombier committed:
Removing irrelevant assertion

Swiftb0y committed:
Merge pull request #13650 from daschuer/gh13646

Fix writing global key text after analyzing to library and file metadata

acolombier committed:
Fix invalid assert and add default out of range usecase

ywwg committed:
Merge pull request #13393 from acolombier/fix/s4mk3-motor

Kontrol S4 Mk3: rework jogwheel speed compute and motorized platter

Swiftb0y committed:
Merge pull request #13656 from acolombier/fix/controller-setting-integer-negative-range

fix: allow controller setting integer to have a negative range

ronso0 committed:
Library: right-align track number column

daschuer committed:
Merge pull request #13339 from ronso0/beats-undo-skin-helper

Skins: add 'beats undo' button

ronso0 committed:
Tracks: allow searching for track id

daschuer committed:
Merge pull request #13694 from ronso0/search-track-id

Tracks: allow searching for track id

ronso0 committed:
Merge remote-tracking branch 'mixxx/2.4' into 2.5

uklotzde committed:
Fix replay gain analysis

Holzhaus committed:
Merge pull request #13703 from uklotzde/replaygain

Fix replay gain analysis

ekordas committed:
MIDI for light:  add settings ui

ekordas committed:
Change Label Names to be more readable

ekordas committed:
MIDI for light: Regroup Settings Layout

ronso0 committed:
ColorSchemeParser: move node finder to separate function

ronso0 committed:
LegacySkinParser: look for LaunchImage style in configured color scheme

ronso0 committed:
LateNight/Shade: adjust LaunchImage style for color schemes

daschuer committed:
Merge pull request #13455 from ronso0/autodj-show-decks34

AutoDJ: force-show decks 3/4 if we are going to use them

ronso0 committed:
(fix) Controllers: return from loadSettings() when there's no mapping file

ronso0 committed:
Show wait cursor when re/loading a skin (not during startup)

daschuer committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

JoergAtGithub committed:
Fix additional "c++11 range-loop might detach Qt container (QList) [-Wclazy-range-loop-detach]" errors in 2.5 branch

JoergAtGithub committed:
Fix "Use QLatin1String("") or QString() instead of an empty QStringLiteral [-Wclazy-empty-qstringliteral]" error

JoergAtGithub committed:
Merge pull request #13738 from ronso0/mapping-file-doesnt-exist

(fix) Controllers: return from loadSettings() when there's no mapping file

JoergAtGithub committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

daschuer committed:
Merge pull request #13769 from JoergAtGithub/FixClazy12errors_2.5

Fix Clazy v1.12 errors in 2.5

daschuer committed:
Merge pull request #13747 from ronso0/skin-reload-wait-cursor

Show wait cursor when re/loading a skin (not during startup)

JoergAtGithub committed:
Reduced include scope from module include of QtQml by class include of QQmlEngine

JoergAtGithub committed:
Removed module wide include of QtSql and added class includes where necessary

JoergAtGithub committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

JoergAtGithub committed:
Removed module wide include of QtConcurrent and added sub includes where necessary

JoergAtGithub committed:
Removed module wide include of QtDbus and added class includes instead

daschuer committed:
Merge pull request #13788 from JoergAtGithub/removeQtModuleHeaderIncludes_2.5

Replace module wide Qt includes to make 2.5 compliant with Clazy 1.12

JoergAtGithub committed:
Merge pull request #13731 from ronso0/launchimagestyle-per-color-scheme

(skins) allow to set LaunchImage style per color scheme

ronso0 committed:
LateNight: use Classic launch image style also for 64 samplers version

JoergAtGithub committed:
Merge pull request #13796 from ronso0/late64-launchimage

LateNight: use Classic launch image style also for 64 samplers version

JoergAtGithub committed:
Merge remote-tracking branch 'upstream/2.5' into SlowEventLoopNotify

daschuer committed:
Merge pull request #13674 from ronso0/lib-right-align-dates

Library: center date values, right-align Track #

JoergAtGithub committed:
Merge remote-tracking branch 'upstream/2.5' into SlowEventLoopNotify

JoergAtGithub committed:
Adjusted warning threshold again to 10ms

Commit:
add missing header

JoergAtGithub committed:
Merge pull request #13825 from m0dB/fix-compilation-2.5-mac

add missing header, fixes compilation of 2.5 on macOS

daschuer committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

JoergAtGithub committed:
Update 2.5 buildenv to latest version

daschuer committed:
Merge pull request #13830 from JoergAtGithub/update_buildenv_2.5

Update 2.5 buildenv to latest version

daschuer committed:
Merge pull request #12094 from JoergAtGithub/SlowEventLoopNotify

Add a debug message, which appears when event loop processing in Mixxx application takes very long

Holzhaus committed:
fix(build): Fix `-Wdeprecated-declarations` warnings for `QCheckBox`

Fixes warnings such as:

    mixxx/src/controllers/legacycontrollersettings.cpp: In member function ‘virtual QWidget* LegacyControllerBooleaetting::buildInputWidget(QWidget*)’:
    mixxx/src/controllers/legacycontrollersettings.cpp:144:36: error: ‘void QCheckBox::stateChanged(int)’ is depreced: Use checkStateChanged() instead [-Werror=deprecated-declarations]
      144 |     connect(pCheckBox, &QCheckBox::stateChanged, this, [this](int state) {
          |                                    ^~~~~~~~~~~~
    In file included from /usr/include/qt6/QtWidgets/QCheckBox:1,
                     from mixxx/src/controllers/legacycontrollersettings.cpp:5:/usr/include/qt6/QtWidgets/qcheckbox.h:41:10: note: declared here
       41 |     void stateChanged(int);
          |          ^~~~~~~~~~~~
    cc1plus: all warnings being treated as errors

Holzhaus committed:
fix(build): Fix `-Wdeprecated-declarations` warnings for `QDateTime`

Fixes warnings such as:

    In file included from mixxx/src/library/browse/browsethread.cpp:10:
    mixxx/src/util/datetime.h: In function ‘QDateTime mixxx::localDateTimeFromUtc(const QDateTime&)’:
    mixxx/src/util/datetime.h:16:51: error: ‘QDateTime::QDateTime(QDate, QTime, Qt::TimeSpec, int)’ is deprecated:
    Pass QTimeZone instead [-Werror=deprecated-declarations]
       16 |     return QDateTime(dt.date(), dt.time(), Qt::UTC).toLocalTime();
          |                                                   ^
    In file included from /usr/include/qt6/QtCore/QDateTime:1,
                     from mixxx/src/util/fileinfo.h:5,
                     from mixxx/src/util/fileaccess.h:3,
                     from mixxx/src/library/browse/browsethread.h:14,
                     from mixxx/src/library/browse/browsethread.cpp:1:
    /usr/include/qt6/QtCore/qdatetime.h:359:5: note: declared here
      359 |     QDateTime(QDate date, QTime time, Qt::TimeSpec spec, int offsetSeconds = 0);
          |     ^~~~~~~~~
    cc1plus: all warnings being treated as errors

Holzhaus committed:
fix(build): Fix `-Wdeprecated-declarations` warnings for `QDomDocument`

Fixes warnings such as:

    mixxx/src/skin/legacy/legacyskinparser.cpp: In static member function ‘static QDomE
    lement LegacySkinParser::openSkin(const QString&)’:
    mixxx/src/skin/legacy/legacyskinparser.cpp:237:25: error: ‘bool QDomDocument::setCo
    ntent(QIODevice*, QString*, int*, int*)’ is deprecated: Use the overload returning ParseResult instead
    . [-Werror=deprecated-declarations]
      237 |     if (!skin.setContent(&skinXmlFile,&errorMessage,&errorLine,&errorColumn)) {
          |          ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    In file included from /usr/include/qt6/QtXml/QDomElement:1,
                     from mixxx/src/skin/legacy/legacyskinparser.h:3,
                     from mixxx/src/skin/legacy/legacyskinparser.cpp:1:
    /usr/include/qt6/QtXml/qdom.h:330:10: note: declared here
      330 |     bool setContent(QIODevice *dev, QString *errorMsg, int *errorLine = nullptr, int *errorCol
    umn = nullptr);
          |          ^~~~~~~~~~

ronso0 committed:
(fix) skin reload: store new color scheme

Holzhaus committed:
refactor: Use correct parameter value for `checkStateChanged` slots

Holzhaus committed:
refactor(weffectchainpresetbutton): Deduplicate some code

Holzhaus committed:
refactor(xml): Use `QString::number` to avoid multi-`arg`

daschuer committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

daschuer committed:
Make iterator const

Swiftb0y committed:
Merge pull request #13845 from Holzhaus/qt6.7-deprecated-decls

Qt 6.8 deprecated declaration fixes

Holzhaus committed:
Merge pull request #13856 from daschuer/2.5-clazy-ci-const-iterator

Make iterator const

Swiftb0y committed:
Merge pull request #13847 from ronso0/skin-colorscheme-switch-apply

(fix) skin reload: store new color scheme

acolombier committed:
fix: prevent double free on DigitsRenderer

Commit:
use atomic to fix tsan detected data race condition

Commit:
use atomics to fix tsan detected race condition in vsyncthread PLL

Commit:
use atomic for visualplayposition m_valid

daschuer committed:
Merge pull request #13876 from m0dB/tsan-fix-13869

fix tsan issue 13869 visualplayposition

daschuer committed:
Merge pull request #13875 from m0dB/tsan-fix-controlindicator

use atomic to fix tsan detected data race condition of blink value in control indicator

daschuer committed:
Merge pull request #13873 from m0dB/tsan-fix-pll-vars

tsan fix pll vars data race

daschuer committed:
Fix crash due to dangling pointer access in developer mode.

daschuer committed:
Prefix event pointers with p

JoergAtGithub committed:
Merge pull request #13889 from daschuer/gh13885

Fix crash due to dangling pointer access in developer mode.

Swiftb0y committed:
Merge pull request #13890 from fwcd/objc-block-indent-2.5

[2.5] clang-format: Indent Objective-C blocks with 4 spaces

Commit:
use atomic for m_bWakeScheduler, protect m_bQuit with mutex

Commit:
avoid data race on m_pStream

Commit:
act on review

daschuer committed:
Also skip detailed info in case of QEvent::ChildRemoved

ronso0 committed:
(fix) Library > Analysis: enable Stop button when anaylzing crate/playlist

ronso0 committed:
Merge pull request #13900 from daschuer/gh13885

Fix crash in case of QEvent::ChildRemoved

fwcd committed:
MixxxApplication: Check for nullptr targets

Swiftb0y committed:
Merge pull request #13902 from ronso0/anaylysis-stop-button

(fix) Library > Analysis: enable Stop button when anaylzing crate/playlist

daschuer committed:
Merge pull request #13898 from m0dB/tsan-fix-13893

use atomic for m_bWakeScheduler, protect m_bQuit with mutex

daschuer committed:
Merge pull request #13903 from fwcd/handle-notify-target-null

MixxxApplication: Handle nullptr targets in notify

ronso0 committed:
(fix) Track property editor: don't close when opening context menu

fwcd committed:
AudioUnitManager: Initialize isInstantiated

daschuer committed:
Merge pull request #13938 from fwcd/initialize-is-instantiated

AudioUnitManager: Avoid undefined behavior by initializing `m_isInstantiated` to `false`

daschuer committed:
Merge pull request #13930 from ronso0/track-prop-edi-menu-fix

(fix) Track property editor: don't close when opening context menu

daschuer committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

daschuer committed:
Update Translation template. Found 3111 source text(s) (11 new and 3100 already existing)

daschuer committed:
Pull latest translations from https://www.transifex.com/mixxx-dj-software/mixxxdj/brmixxx2-5 Compile QM files out of TS files that are used by the localized app

fwcd committed:
MC7000: Add sampler options to mapping settings

Commit:
atomic m_pStream and atomic return value

JoergAtGithub committed:
Merge pull request #13950 from fwcd/mc7000-sampler-config

Denon MC7000: Add sampler options to mapping settings

daschuer committed:
Merge pull request #13899 from m0dB/tsan-fix-pa-stream-warnings

avoid data race on m_pStream

daschuer committed:
Pull latest translations from https://www.transifex.com/mixxx-dj-software/mixxxdj/mixxx2-5/. Compile QM files out of TS files that are used by the localized app

daschuer committed:
Merge remote-tracking branch 'upstream/2.4' into 2.5

daschuer committed:
Update CHANGLOG for 2.5.0

ronso0 committed:
Merge pull request #13189 from daschuer/gh13030

Revert "DEBUG output for fullscreen changes"

daschuer committed:
Merge pull request #13953 from m0dB/until_mark_text_height_limit-2.5

limit play marker hints text height to 1/3rd or full waveform height (2.5)

Commit:
resolveTrackIdsFromUrls directly, without going through DragAndDropeHlper::supportedTracksFromUrls

daschuer committed:
Merge pull request #13958 from mixxxdj/resolve-from-urls-2.5

resolveTrackIdsFromUrls directly, without going through supportedTracksFromUrls

fwcd committed:
EngineRecord: Fix recorded duration at start of recording

Swiftb0y committed:
Merge pull request #13966 from fwcd/fix-recorded-duration

EngineRecord: Fix recorded duration at start of recording

ronso0 committed:
Controller settings: have clickable checkbox label (like QCheckBox)

acolombier committed:
Merge pull request #13979 from ronso0/controller-sett-clickable-checkbox-label

Controller settings: have clickable checkbox label (like QCheckBox)

ronso0 committed:
Preferences: improve setScrollSafeGuardForAllInputWidgets(), add p prefix

ronso0 committed:
Controller preferences: add scrollSafeGuard for settings widgets

ronso0 committed:
(fix) preferences: send scroll event to page widget when hitting an input widget while scrolling

Swiftb0y committed:
Merge pull request #13990 from ronso0/controller-sett-scrollSafeGuard

(fix/UX) set scrollSafeGuard also for controller settings widgets

JoergAtGithub committed:
Merge pull request #13999 from mixxxdj/2.4

Merge 2.4 into 2.5

ronso0 committed:
Merge pull request #13956 from daschuer/changelog_25

Changelog for 2.5.0

fwcd committed:
Merge pull request #13721 from ekordas/MIDI-for-light-add-settings-ui

MIDI for light:  add settings ui

ronso0 committed:
(fix) also exclude QEvent::Timer in MixxxApplication::notify()

daschuer committed:
Merge pull request #14012 from ronso0/notify-fix

(fix) also exclude QEvent::Timer in MixxxApplication::notify()

Commit:
insert width and height in svgs where missing

ronso0 committed:
Merge pull request #14013 from m0dB/fix-deere-svgs

insert width and height in svgs where missing

daschuer committed:
Test for empty key text first

daschuer commented:
This one is obsolete because an alternative has already been merged.
